x-pack/plugins/actions/server/sub_action_framework/sub_action_connector.ts

```diff
@@ -133,27 +133,19 @@ export abstract class SubActionConnector<Config, Secrets> {
       this.ensureUriAllowed(url);
       const normalizedURL = this.normalizeURL(url);
```

Comment:
I am wondering if the validation and assertion of the URL (`this.assertURL`, `this.ensureUriAllowed`, and `this.normalizeURL`) should be part of the request axios utility function. I am not sure though if it will introduce breaking changes. @ymao1 What do you think?

Reply:
Since the `request` function is used inside the action executors and these validation functions look to be slightly different from the validation functions that we use elsewhere, there is the possibility of breaking existing actions by moving them inside the request function. I'm thinking a followup issue would make sense where we consolidate all of the different things we do to parse and validate urls into utility functions with customizable error messages wdyt? I think we started to do some of this in `actions_config` but I think we could do more.

Reply:
Thanks, @ymao1! Yes, a follow up issue makes more sense. I think that validating the URL in the request utility function will help. As it is, the developer of the connector has to do it manually on each connector and it is easy to forget.